dev/com.ibm.ws.security.mp.jwt_fat/fat/src/com/ibm/ws/security/mp/jwt/fat/MPJwtConfigUsingBuilderTests.java

```diff
@@ -0,0 +1,1087 @@
+     * and appropriate error messages in the server side log
+     * 
+     * @throws Exception
+     */
+    @Test
+    @ExpectedFFDC({ "com.ibm.websphere.security.jwt.InvalidClaimException", "com.ibm.websphere.security.jwt.InvalidTokenException" })
+    public void MPJwtConfigUsingBuilderTests_Audience_Mismatch() throws Exception {
+
+        super.restoreTestServers();
+
+        String builtToken = getJwtTokenUsingBuilder(jwtBuilderServer, "audience_mismatch");
+
+        Expectations expectations = new Expectations();
+        expectations.addExpectation(new ResponseStatusExpectation(testAction, HttpServletResponse.SC_UNAUTHORIZED));
+        expectations.addExpectation(new ServerMessageExpectation(testAction, resourceServer, MpJwtMessageConstants.CWWKS6031E_CAN_NOT_PROCESS_TOKEN, "Messagelog did not contain an error indicating a problem authenticating the request with the provided token."));
+        expectations.addExpectation(new ServerMessageExpectation(testAction, resourceServer, MpJwtMessageConstants.CWWKS6023E_AUDIENCE_NOT_TRUSTED, "Messagelog did not contain an exception indicating that the audience is NOT valid."));
+
+        genericConfigTest(builtToken, expectations);
+
+    }
+
+    /**
+     * Test that uses a builder that puts the aud claim in the JWT Token. The resource server specifies
+     * the audiences config attribute with a superset of the values that are in the token. Expect success
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void MPJwtConfigUsingBuilderTests_Audience_SuperSet() throws Exception {
+
+        super.restoreTestServers();
+
+        String builtToken = getJwtTokenUsingBuilder(jwtBuilderServer, "audience_superset");
+        genericConfigTest(builtToken);
+
+    }
+
+    /**
+     * Test that uses a builder that puts the aud claim in the JWT Token. The resource server specifies
+     * the audiences config attribute with a subset of the values that are in the token. Expect success
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void MPJwtConfigUsingBuilderTests_Audience_Subset() throws Exception {
+
+        super.restoreTestServers();
+
+        String builtToken = getJwtTokenUsingBuilder(jwtBuilderServer, "audience_subset");
+        genericConfigTest(builtToken);
+
+    }
+
+    /******************************** jwksUri/keyName ***************************************/
+
+    // Testing jwksUri and keyName in tandem
+    // The overall SSL config of the builder and mpJwt servers use key/trust that is paired
+    // So, we can use a mix of the following using what would be valid values
+    //  1) (passes) builder issues token with JWK, mpJwt specifies a valid jwksUri
+    //  2) (fails) builder issues token with JWK, mpJwt omits the jwksUri, but, specifies a valid keyName
+    //  3) (passes) builder issues token with x509, mpJwt specifies a valid jwksUri
```

Comment:
You say this case is supposed to pass, but the test itself (`MPJwtConfigUsingBuilderTests_buildUsingX509_mpJWTusingJWK`) is written to be a negative test case.

Reply:
That is a block comment for the 4 tests in the jwksUri/keyName section - I've moved 1-4 to the specific tests